fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/UserAuthentication.java

```diff
@@ -0,0 +1,54 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.analysis;
+
+import org.apache.doris.common.ErrorCode;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.nereids.exceptions.AnalysisException;
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalOlapScan;
+import org.apache.doris.qe.ConnectContext;
+
+/**
+ * Check whether a user is permitted to scan specific tables.
+ */
+public class UserAuthentication extends OneAnalysisRuleFactory {
+
+    @Override
+    public Rule build() {
+        return logicalOlapScan().thenApply(ctx -> checkPermission(ctx.root, ctx.connectContext))
```

Comment:
we should use logicalRelation pattern, but i checked GeneratedPatterns and we have no logicalRelation pattern. So i think we need to update pattern generator to generate it. @924060929 could we do it in pattern generator?

Reply:
`logicalRelation` is included in [fe/fe-core/src/main/java/org/apache/doris/nereids/pattern/Patterns.java#L172](https://github.com/apache/doris/blob/master/fe/fe-core/src/main/java/org/apache/doris/nereids/pattern/Patterns.java#L172).